query/src/sql/planner/semantic/type_check.rs

```diff
@@ -899,4 +905,47 @@ impl<'a> TypeChecker<'a> {
             data_type,
         ))
     }
+
+    async fn resolve_position(
+        &self,
+        substr_expr: &'_ Expr<'a>,
+        str_expr: &'_ Expr<'a>,
+    ) -> Result<(Scalar, DataTypeImpl)> {
+        let first_str = self.match_expr(substr_expr)?;
+        let second_str = self.match_expr(str_expr)?;
+        let index = DataValue::UInt64(match second_str.find(first_str.as_str()) {
+            None => 0,
+            Some(idx) => idx + 1,
+        } as u64);
+        Ok((
+            ConstantExpr {
+                value: index.clone(),
+                data_type: index.data_type(),
+            }
+            .into(),
+            index.data_type(),
+        ))
+    }
+
+    fn match_expr(&self, expr: &'_ Expr<'a>) -> Result<String> {
+        let expr_str = match expr {
```

Comment:
For the sql: `SELECT POSITION("xbar" IN "xbar0");`
`"xbar"` will be identified as `Expr::ColumnRef`, is this the correct result for parser? @andylokandy  If so, I need to handle the `ColumnRef` case here.

Reply:
You can replace double quote `"` with single quote `'`.